commcare_connect/opportunity/tasks.py

```diff
@@ -343,3 +344,19 @@ def generate_catchment_area_export(opportunity_id: int, export_format: str):
     export_tmp_name = f"{now().isoformat()}_{opportunity.name}_catchment_area.{export_format}"
     save_export(dataset, export_tmp_name, export_format)
     return export_tmp_name
+
+
+@celery_app.task(bind=True)
+def bulk_update_payments_task(self, opportunity_id: int, headers: list[str], rows: list[list]):
+    from commcare_connect.opportunity.visit_import import ImportException, bulk_update_payments
+
+    set_task_progress(self, "Payment Record Import is in porgress.")
+    try:
+        status = bulk_update_payments(opportunity_id, headers, rows)
+        messages = [f"Payment status updated successfully for {len(status)} users."]
+        if status.missing_users:
+            messages.append(status.get_missing_message())
+    except ImportException as e:
+        messages = [f"Payment Import failed: {e}"] + getattr(e, "invalid_rows", [])
+
+    set_task_progress(self, f"<br>".join(messages), is_complete=True)
```

Comment:
how do these messages get shown to the user? are they maintained if the user navigates away?